src/util/invariant.h

```diff
@@ -254,14 +268,21 @@ inline void invariant_violated_string(
 // This condition should be used to document that assumptions that are
 // made on goto_functions, goto_programs, exprts, etc. being well formed.
 // "The data structure is corrupt or malformed"
+//
+// It is okay for a DATA_INVARIANT to have an empty string as a reason string
+// if you can't state the reason, or doing so doesn't add any information -
+// just restates the condition.
```

Comment:
IMHO doesn't belong as part of the documentation, also note that there are almost always better options.